spec/graphql/fragment_cache/cache_key_builder_spec.rb

```diff
@@ -111,4 +112,27 @@
 
     specify { is_expected.to eq "schema_key/cachedPost(id:#{id})[id.title]/#{post.cache_key}/custom/99" }
   end
+
+  context "when context_key is passed" do
+    let(:user) { CacheableUser.new(id: 33, name: "John") }
+    let(:context) { {user: user} }
+    let(:context_key) { :user }
+    subject { described_class.call(object: object, query: query_obj, path: path, context_key: context_key) }
+
+    specify { is_expected.to eq "schema_key/cachedPost(id:#{id})[id.title]/#{user.cache_key}" }
+
+    context "when context_key is array" do
+      let(:app_version) { 3 }
+      let(:context) { {user: user, app_version: app_version} }
+      let(:context_key) { [:user, :app_version] }
+
+      specify { is_expected.to eq "schema_key/cachedPost(id:#{id})[id.title]/#{user.cache_key}/#{app_version}" }
+
+      context "when context value contains nil" do
+        let(:user) { nil }
+
+        specify { is_expected.to eq "schema_key/cachedPost(id:#{id})[id.title]//#{app_version}" }
```

Comment:
Maybe we should return `"nil"` from `nil._graphql_cache_key` instead of `""` to make this key look `schema_key/cachedPost(id:#{id})[id.title]/nil/#{app_version}`